handbook/engineering/code-intelligence/goals.md

```diff
@@ -0,0 +1,41 @@
+# Code Intelligence goals and priorities
+
+## Goals
+
+Progress on adoption and usage is tracked in our [Looker dashboard](https://sourcegraph.looker.com/dashboards/131).
+
+### Grow LSIF adoption and usage reaching 10k precise code intel WAUs
+
+**Problem:** By default, users get basic code intel results which are often incorrect for tokens with common names (such as Get). This is solved by enabling precise code intel, which gives users faster, 100% accurate results for a definition or reference for a symbol they hover over. 
+
+**Strategy:** It is imperative that we raise adoption *deeply* within a particular language and ecosystem. We want to provide fast and precise answers about a user's _universe of code_. This can only happen when that entire universe is indexed. We intend to provide an index for _all_ such universes of code by making progress on two fronts. First, we need to raise excitement for the opportunities adopting precise code intelligence makes possible. Once a user sees what is possible, they should have little reason not to put in the effort of adoption. Second, we need to reduce the friction felt during indexing (especially during adoption). Users tend to perceive only the work we ask them to do to set up precise code intelligence and do not perceive the full benefit of precise code intelligence. Reducing the friction of indexing is a critical step in growing support. It is also necessary to raise adoption *widely* across companies. Until we have support for a set of indexers that aligns closely with a company's tech stack, we cannot pursue those companies for adoption. There will always be a large pool of untapped users we can reach by adding support for additional languages.
+
+**Milestones:**
+
+1. Validate Go support: Reach 80% precise code intelligence for Go within 3 organizations.
```

Comment:
What does 80% mean here? We need this to be measurable in a dashboard. My proposal is that we need to have a concrete WAU number for each of these milestones (and the sum should add up to at least 10k).

Reply:
In other words, answer the question: "how many WAU do we think each of these milestones will bring us"?

Reply:
> We need this to be measurable in a dashboard.

This _is_ measurable in a dashboard. We break out code intel actions between lsif and search based. If we only fall back 20% of the time we've met our success criteria.

Reply:
Is your suggestion to say that milestones contributes to WAU or each customer contributes to WAU? In the former case we need to know how in-demand a language is and in the latter we need to know what the number of potential users of Sg are at that company.

Reply:
This would be measurable as "X% of Go code intel queries are precise at a certain customer" and are working to get this data on the dashboard.

Reply:
We currently have this data, but not broken down by language. We will have this broken down by the end of this milestone.

Reply:
Maria's wording is clearer.
```suggestion
1. Validate Go support: 80% of Go code intel queries are precise within 3 customers.
```

But, even with this updated wording, it doesn't connect the dots for me.

What needs to be "validated"? Do we expect achieving this milestone with directly increase WAU at these customers (or is there some other reason we are doing this)? If WAU is the point of this milestone, then it seems like we should be able to estimate/forecast how much we expect each of these actions to impact WAU.

For example (I am just making up numbers):

- [Deliver to $U](https://github.com/sourcegraph/customer/issues/78) +1111 WAU
- [Deliver to $S](https://github.com/sourcegraph/customer/issues/77) +2222 WAU
- [Deliver to $C](https://github.com/sourcegraph/customer/issues/76) +3333 WAU

If we do this for all the milestones, then we can "show our work" on how we got to the top level goal (which is currently 10k, a made up number).